BackEnd/routes/points.py

```
@@ -0,0 +1,126 @@
+    Add points to a member by name. Creates member if they don't exist.
+    This is the endpoint the frontend uses for the simple "Add Points" form.
+    """
+    try:
+        # Validate action type
+        try:
+            action_type = ActionType(request.action)
+        except ValueError:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail=f"Invalid action type. Must be one of: {[e.value for e in ActionType]}"
+            )
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)